src/client/components/modals/AfterSave.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Text } from '@aragon/ui'
+
+import { ModalWrapper } from './ModalWrapper'
+import {
+  CheckWrapper,
+  IconSuccess,
+  IconError,
+  AnimationLoadingCircle,
+} from '../styled-components'
+
+const calculateText = (
+  savingProfile,
+  savedProfileSuccess,
+  savedProfileError
+) => {
+  if (savingProfile) return 'Updating your profile...'
+  else if (savedProfileSuccess) return 'Your profile has been updated!'
+  else if (savedProfileError) return 'Your profile could not be updated.'
```

Comment:
Not sure if it's cleaner or not, but this function could be replaced with an object.

```js
const messages = {
  savingProfile: 'Updating your profile...',
  savedProfileSuccess: 'Your profile has been updated!',
  savedProfileError: 'Your profile could not be updated.',
}
```

Then, instead of passing boolean props into `AfterSave`, pass `modal.type` directly, called `mode` or something in this component. Then `messages` can be used like this:

```jsx
<Text size="xxlarge">
  {messages[mode] || 'Oops, something went wrong on our end. Please try again.'}
</Text>